Index: HLT/libAliHLTTPC.pkg
===================================================================
--- HLT/libAliHLTTPC.pkg	(revision 34701)
+++ HLT/libAliHLTTPC.pkg	(working copy)
@@ -4,6 +4,7 @@
 TRACKING_CA:=	tracking-ca/AliHLTTPCCATrack.h \
 		tracking-ca/AliHLTTPCCATracklet.h \
 		tracking-ca/AliHLTTPCCATracker.h \
+		tracking-ca/AliHLTTPCCATrackerFramework.h \
 		tracking-ca/AliHLTTPCCAOutTrack.h  \
 		tracking-ca/AliHLTTPCCATrackParam.h \
 		tracking-ca/AliHLTTPCCATrackConvertor.h \
@@ -121,7 +122,7 @@
 # 		tracking/AliHLTTPCHistogram1D.h \
 # 		tracking/AliHLTTPCHistogramAdaptive.h
 
-MODULE_SRCS= 	$(CLASS_HDRS:.h=.cxx) \
+MODULE_SRCS:= 	$(CLASS_HDRS:.h=.cxx) \
 		AliHLTTPCLog.cxx
 
 MODULE_HDRS:= 	$(CLASS_HDRS) \
@@ -150,6 +151,21 @@
 		   RAW \
 		   STEER 
 
+#GPU Tracker Interface		   
+CLASS_HDRS	+= tracking-ca/AliHLTTPCCAGPUTracker.h
+
+#GPU Tracker Build enabled
+ELIBS		:= cudart
+ELIBSDIR	:= /opt/cuda/lib64
+EDEFINE		:= BUILD_GPU
+MODULE_CUHDRS	+= $(TRACKING_CA) tracking-ca/AliHLTTPCCAGPUTracker.h
+MODULE_CUSRCS	+= tracking-ca/AliHLTTPCCAGPUTracker.cu
+
+#GPU Tracker Build disabled
+#MODULE_SRCS	+= tracking-ca/AliHLTTPCCAGPUTrackerDummy.cxx
+
+		
+
 LIBRARY_DEP := -lHLTbase -lAliHLTRCU -lAliHLTUtil \
                -lCDB -lESD -lSTEER -lSTEERBase -lTPCrec -lTPCcalib -lTPCbase \
                -lRAWDatarec -lRAWDatabase \
@@ -162,8 +178,10 @@
 include $(MODDIR)/hlt.conf
 
 SRCS:=$(patsubst %,TPCLib/%,$(MODULE_SRCS))
+CUSRCS:=$(patsubst %,TPCLib/%,$(MODULE_CUSRCS))
 CINTHDRS:=$(patsubst %,TPCLib/%,$(CLASS_HDRS))
 HDRS:=$(patsubst %,TPCLib/%,$(MODULE_HDRS))
+CUHDRS:=$(patsubst %,TPCLib/%,$(MODULE_CUHDRS))
 DHDR:=$(patsubst %,TPCLib/%,$(MODULE_DHDR))
 CINTAUTOLINK:= $(shell test "x$(MODULE_DHDR)" = "x" && echo 1)
 
Index: HLT/CMake_libAliHLTTPC.txt
===================================================================
--- HLT/CMake_libAliHLTTPC.txt	(revision 34701)
+++ HLT/CMake_libAliHLTTPC.txt	(working copy)
@@ -80,6 +80,8 @@
 TPCLib/tracking-ca/AliHLTTPCCATrack.cxx
 TPCLib/tracking-ca/AliHLTTPCCATracklet.cxx
 TPCLib/tracking-ca/AliHLTTPCCATracker.cxx
+TPCLib/tracking-ca/AliHLTTPCCATrackerFramework.cxx
+TPCLib/tracking-ca/AliHLTTPCCAGPUTracker.cxx
 TPCLib/tracking-ca/AliHLTTPCCAOutTrack.cxx
 TPCLib/tracking-ca/AliHLTTPCCATrackParam.cxx
 TPCLib/tracking-ca/AliHLTTPCCATrackConvertor.cxx
